### PR TITLE
Spreadsheet: Stuff and things

### PR DESCRIPTION
**Pull request by alimpfard**
- Don't crash when deleting chars from the formula editor (works around the main issue in #14043)
- `` R`sheet("foo"):A0:A69` ``
- Make `lookup` throw instead of returning `undefined` when it fails to find something (even returning `undefined` is better than what Excel and friends do - returning the wrong thing!)
- "Import sheets from..." option for all your multiple CSV importing needs
- Error stack traces!

I meant to do more with this, but some weird HashMap problem got me bored and I stopped here ^^'

### CODE REVIEW

File: Userland/Applications/Spreadsheet/SpreadsheetModel.cpp [hunk 1]
```diff
@@ -103,6 +103,36 @@ GUI::Variant SheetModel::data(const GUI::ModelIndex& index, GUI::ModelRole role)
...
+    if (to_underlying(role) == to_underlying(Role::Tooltip)) {
+        auto const* cell = m_sheet->at({ (size_t)index.column(), (size_t)index.row() });
+        if (!cell || !cell->thrown_value().has_value())
+            return {};
+
+        auto value = cell->thrown_value().value();
+        if (value.is_object()) {
+            auto& object = value.as_object();
+            if (is<JS::Error>(object)) {
+                auto& error = static_cast<JS::Error&>(object);
```
Comment (linusg) [inline Userland/Applications/Spreadsheet/SpreadsheetModel.cpp:115]: could use early return style here to make it a bit nicer

```cpp
if (!value.is_object())
    return {};
auto& object = value.as_object();
if (is<JS::Error>(object))
    return {};
auto& error = static_cast<JS::Error&>(object);
...
```

**Review by linusg — COMMENTED** [review]
> AK: Recognise

*Sad en-US noises*

File: Userland/Applications/Spreadsheet/SpreadsheetModel.cpp (hunk 1 above)
Comment (alimpfard) [reply Userland/Applications/Spreadsheet/SpreadsheetModel.cpp:115]: It's a copy of earlier code, but yeah, I like this better, will fix the old one too.

**Comment by alimpfard** [comment]
(also there are failing tests, will look at fixing them later)

**Comment by alimpfard** [comment]
> Sad en-US noises

Made en-US happy too

**Comment by hjalves** [comment]
I've tried this PR and it worked pretty well. 

It would be nice if we could also reference cells in other sheets, without using ranges, like `=sheet("foo"):A0`.

**Comment by alimpfard** [comment]
>It would be nice if we could also reference cells in other sheets, without using ranges, like `=sheet("foo"):A0`.
you can already do that, `sheet("foo").A0`.

**Review by linusg — APPROVED** [review]
One GBP for very good PR title

**Pull request merged**